common.blocks/sidebar/__sorting/sidebar__sorting.js

```diff
@@ -13,16 +13,17 @@ modules.define(
         },
 
         _onSelectChange: function() {
-            var sortType = this._select.getVal();
+            var sort = this._select.getVal();
             var direction = 'desc';
 
-            if (sortType.indexOf('-reverse') !== -1) {
-                sortType = sortType.split('-')[0];
+            if (sort.indexOf('-reverse') !== -1) {
+                sort = sort.split('-')[0];
                 direction = 'asc';
             }
 
-            Location.change({ params: this._getParams(sortType, direction) });
-            window.location.reload();
+            Location.change({ params: this._getParams(sort, direction) });
```

Comment:
@tadatuta can you provide arguments why subscribing to location change is better?